pkl-core/src/main/java/org/pkl/core/util/HttpUtils.java

```diff
@@ -47,4 +48,22 @@ public static void checkHasStatusCode200(HttpResponse<?> response) throws IOExce
     throw new IOException(
         ErrorMessages.create("badHttpStatusCode", response.statusCode(), response.uri()));
   }
+
+  public static URI setPort(URI uri, int port) {
+    if (port < 0 || port > 65535) {
+      throw new IllegalArgumentException(String.valueOf(port));
+    }
+    try {
+      return new URI(
+          uri.getScheme(),
+          uri.getUserInfo(),
+          uri.getHost(),
+          port,
+          uri.getPath(),
+          uri.getQuery(),
+          uri.getFragment());
+    } catch (URISyntaxException e) {
+      throw new AssertionError(e); // only port changed
```

Comment:
```suggestion
      throw PklBugException.unreachableCode(); // only port changed
```

Reply:
Done. `org.pkl.core.util` depending on `org.pkl.core` seems problematic. But since `PklBugException` is already used in this class, it's not making things worse.